src/shared/clients/defaultCloudWatchLogsClient.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { CloudWatchLogs } from 'aws-sdk'
+import { ext } from '../extensionGlobals'
+import { CloudWatchLogsClient } from './cloudWatchLogsClient'
+
+export class DefaultCloudWatchLogsClient implements CloudWatchLogsClient {
+    public constructor(public readonly regionCode: string) {}
+
+    public async *describeLogGroups(): AsyncIterableIterator<CloudWatchLogs.LogGroup> {
+        const sdkClient = await this.createSdkClient()
+        const request: CloudWatchLogs.DescribeLogGroupsRequest = {}
+        do {
+            const response = await this.invokeDescribeLogGroups(request, sdkClient)
+            if (response.logGroups) {
+                yield* response.logGroups
+            }
+            request.nextToken = response.nextToken
+        } while (request.nextToken)
+    }
+
+    protected async invokeDescribeLogGroups(
+        request: CloudWatchLogs.DescribeLogGroupsRequest,
+        sdkClient: CloudWatchLogs
+    ): Promise<CloudWatchLogs.DescribeLogGroupsResponse> {
+        return sdkClient.describeLogGroups(request).promise()
+    }
+
+    protected async createSdkClient(): Promise<CloudWatchLogs> {
```

Comment:
Consider providing the SDK client in the constructor. It lets you mock the Cloudwatch impl without having to change global state via `ext.sdkClientBuilder`.

Reply:
Interesting. Mind if that waits for a future PR? Would like to one-fell-swoop that across other clients.